contracts/UniswapV3Staker.sol

```diff
@@ -222,29 +222,31 @@ contract UniswapV3Staker is
         Stake memory stake = stakes[params.tokenId][incentiveId];
 
         require(stake.exists == true, 'nonexistent stake');
-        require(incentive.rewardToken != address(0), 'incentive not found');
 
-        (uint128 reward, uint160 secondsInPeriodX128) =
-            _getRewardAmount(
-                stake,
-                incentive,
-                params,
-                poolAddress,
-                tickLower,
-                tickUpper
-            );
-
-        incentives[incentiveId].totalSecondsClaimedX128 += secondsInPeriodX128;
-
-        // TODO: is SafeMath necessary here? Could we do just a subtraction?
-        incentives[incentiveId].totalRewardUnclaimed = uint128(
-            SafeMath.sub(incentive.totalRewardUnclaimed, reward)
-        );
-
-        // Makes rewards available to claimReward
-        rewards[incentive.rewardToken][msg.sender] = uint128(
-            SafeMath.add(rewards[incentive.rewardToken][msg.sender], reward)
-        );
+        // if incentive still exists
+        if (incentive.totalRewardUnclaimed > 0) {
+          (uint128 reward, uint160 secondsInPeriodX128) =
+              _getRewardAmount(
+                  stake,
+                  incentive,
+                  params,
+                  poolAddress,
+                  tickLower,
+                  tickUpper
+              );
+
+          incentives[incentiveId].totalSecondsClaimedX128 += secondsInPeriodX128;
+
+          // TODO: is SafeMath necessary here? Could we do just a subtraction?
+          incentives[incentiveId].totalRewardUnclaimed = uint128(
+              SafeMath.sub(incentive.totalRewardUnclaimed, reward)
+          );
+
+          // Makes rewards available to claimReward
+          rewards[incentive.rewardToken][msg.sender] = uint128(
+              SafeMath.add(rewards[incentive.rewardToken][msg.sender], reward)
+          );
+        }
 
         delete stakes[params.tokenId][incentiveId];
         emit TokenUnstaked(params.tokenId, incentiveId);
```

Comment:
Wndering whether adding rewards to this event makes sense..... Thinking of a situation where someone's confused as to why no rewards came from the unstake.

Reply:
I don't feel strongly either way. Let's do it as a separate PR if we do decide to.